tools/Customization/DevHome.Customization/ViewModels/DevDriveInsightsViewModel.cs

```diff
@@ -0,0 +1,605 @@
+    [ObservableProperty]
+    private bool _devDriveLoadingCompleted;
+
+    [ObservableProperty]
+    private bool _devDriveOptimizerLoadingCompleted;
+
+    [ObservableProperty]
+    private bool _devDriveOptimizedLoadingCompleted;
+
+    [ObservableProperty]
+    private AdvancedCollectionView _providerComboBoxNamesCollectionView;
+
+    public ObservableCollection<string> DevDrivesSortOptions { get; private set; }
+
+    public AdvancedCollectionView DevDrivesCollectionView { get; private set; }
+
+    public AdvancedCollectionView DevDriveOptimizersCollectionView { get; private set; }
+
+    public AdvancedCollectionView DevDriveOptimizedCollectionView { get; private set; }
+
+    public IDevDriveManager DevDriveManagerObj { get; private set; }
+
+    public string SelectedDevDriveProviderComboBoxName { get; set; }
+
+    private IEnumerable<IDevDrive> ExistingDevDrives { get; set; } = Enumerable.Empty<IDevDrive>();
+
+    public DevDriveInsightsViewModel(
+        ISetupFlowStringResource stringResource,
+        SetupFlowViewModel setupflowModel,
+        SetupFlowOrchestrator orchestrator,
+        IDevDriveManager devDriveManager,
+        DevDriveViewModelFactory devDriveViewModelFactory,
+        DevDriveOptimizerViewModelFactory devDriveOptimizerViewModelFactory,
+        DevDriveOptimizedViewModelFactory devDriveOptimizedViewModelFactory,
+        ToastNotificationService toastNotificationService)
+        : base(stringResource, orchestrator)
+    {
+        // Setup initial state for page.
+        _setupFlowOrchestrator = orchestrator;
+        _setupFlowOrchestrator.CurrentSetupFlowKind = SetupFlowKind.SetupTarget;
+        _setupFlowStringResource = stringResource;
+
+        PageTitle = _setupFlowStringResource.GetLocalized(StringResourceKey.SetupTargetPageTitle);
+        _allKeyWordLocalized = _setupFlowStringResource.GetLocalized(StringResourceKey.SetupTargetAllComboBoxOption);
+
+        // Add the "All" option to the combo box and make sure its always sorted.
+        SelectedDevDriveProviderComboBoxName = _allKeyWordLocalized;
+        _devDriveProviderComboBoxNames = new() { SelectedDevDriveProviderComboBoxName, };
+        ProviderComboBoxNamesCollectionView = new AdvancedCollectionView(_devDriveProviderComboBoxNames, true);
+        ProviderComboBoxNamesCollectionView.SortDescriptions.Add(new SortDescription(SortDirection.Ascending));
+
+        // Add sort options like A-Z, Z-A, etc.
+        DevDrivesSortOptions = new ObservableCollection<string>
+        {
+            _setupFlowStringResource.GetLocalized(StringResourceKey.SetupTargetSortAToZLabel),
+            _setupFlowStringResource.GetLocalized(StringResourceKey.SetupTargetSortZToALabel),
+        };
+
+        // Add AdvancedCollectionView to make filtering and sorting the list of DevDrivesListViewModels easier.
+        DevDrivesCollectionView = new AdvancedCollectionView(_devDriveViewModelList, true);

```

Comment:
I don't understand. Why are you making a view?

Reply:
Yes, I would clean up list of lists.